cmake/modules/hunter_add_host_subdirectory.cmake

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2016 Fabian Winnen
+# All rights reserved.
+
+# Adds a subdirectory with CMakeList.txt as seperate build with default (host) toolchain
+
+# Usage:
+#   hunter_add_host_subdirectory(host)
+#   hunter_add_host_subdirectory(host
+#       BINARY_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/host-build
+#       ENV_VARS CC CXX
+#   )
+
+# Note:
+#   With polly Android toolchain this fails if you do not unset CC and CXX ENV{} variables.
```

Comment:
Should have included the start:

`build.py --clear --toolchain android-ndk-r11c-api-21-arm64-v8a
Python version: 3.5
Build dir: /.../hunter/examples/Protobuf/_builds/android-ndk-r11c-api-21-arm64-v8a
Remove directory: /.../hunter/examples/Protobuf/_builds/android-ndk-r11c-api-21-arm64-v8a
Execute command: [
  `which`
  `cmake`
]

[/.../hunter/examples/Protobuf]> "which" "cmake"

/usr/local/bin/cmake
Execute command: [
  `cmake`
  `--version`
]

[/.../hunter/examples/Protobuf]> "cmake" "--version"

cmake version 3.5.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
Execute command: [
  `cmake`
  `-H.`
  `-B/.../hunter/examples/Protobuf/_builds/android-ndk-r11c-api-21-arm64-v8a`
  `-GUnix Makefiles`
  `-DCMAKE_TOOLCHAIN_FILE=/.../polly/android-ndk-r11c-api-21-arm64-v8a.cmake`
]

[/.../hunter/examples/Protobuf]> "cmake" "-H." "-B/.../hunter/examples/Protobuf/_builds/android-ndk-r11c-api-21-arm64-v8a" "-GUnix Makefiles" "-DCMAKE_TOOLCHAIN_FILE=/Users/Fabian/Development/kinemic/polly/android-ndk-r11c-api-21-arm64-v8a.cmake"

Including HunterGate: /.../hunter/gate/cmake/HunterGate.cmake
-- [polly] Used toolchain: Android NDK r11c / API 21 / arm64-v8a / c++11 support
-- The C compiler identification is GNU 4.9.0
-- The CXX compiler identification is GNU 4.9.0
-- Check for working C compiler: /.../ndk/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64/bin/aarch64-linux-android-gcc
-- Check for working C compiler: /.../ndk/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64/bin/aarch64-linux-android-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /.../ndk/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64/bin/aarch64-linux-android-g++
-- Check for working CXX compiler: /.../ndk/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64/bin/aarch64-linux-android-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- [hunter] invoke cmake subdirectory 'host' build for host...`

I used only polly android toolchain, also i did not find anything in it which could set CC and CXX. Just figured out when inspecting the issue, that those were causing the issue.